api/v1alpha1/argocd_types.go

```diff
@@ -115,6 +115,20 @@ type ArgoCDApplicationControllerShardSpec struct {
 
 	// Replicas defines the number of replicas to run in the Application controller shard.
 	Replicas int32 `json:"replicas,omitempty"`
+
+	// DynamicScalingEnabled defines whether dynamic scaling should be enabled for Application Controller component
+	DynamicScalingEnabled bool `json:"dynamicScalingEnabled,omitempty"`
```

Comment:
Consider making this a pointer to a bool, instead of a bool value. With a pointer, the value can be `nil` in addition to `true` or `false`, which means the option is not set. This allows us to change the default value for this parameter later on.

Reply:
That's a good point!! I will make it a pointer to a bool value.